src/adam_bashforth.jl

```diff
@@ -1,57 +1,76 @@
-module Tmp
 ####################################
-# Explicit Adam-Bashforth solvers
+# Implicit Adam-Bashforth solvers
 ####################################
 # (Hairer & Wanner 1996, Vol I, p.357-358
-include("ODE.jl")
 
-##Fixed step, fixed order, multistep explicit method
 
-function ode_ab(F,y0, tspan,order ::Integer)
-    if !(2 <= order <= 4)
-        error("Currently only orders 2,3, and 4 are implemented for Adam-Bashforth method")
+##Function: ode_imp_ab(F,y0, tspan, order(optional))
+##order is set to 4 by default
+##Adam Bashforth is a fixed step, fixed order, multistep Implicit method
+function ode_imp_ab(F::Function,y0, tspan,order=4 ::Integer)
+    if (0 <= order <= 3)
+        coef = am_imp_coefficients3
+        b = ms_coefficients4
+    else
+        #calculating higher order coefficients for implicit Adam Multon method
+        coef = zeros(order+1, order+1)
```

Comment:
Just use `order` to index into `coef` instead of `k`. i.e. `order==k+1`. Do that too in the main integration loop.